unepwcmc/action-agenda
# Ticket26 form navigation and exit modal

# Summary

Merge #7 (ticket 10) first, as this will probably have conflicts to resolve

## Functionality

The buttons that interact with the survey will currently just `console.log` (the actual methods we'll need to use are provided as comments for now until this is connected up with #7), but the buttons that open modals work

## Other additions/edits

- added `.nvmrc` - not sure if everyone is using 14.18.1 but that version is working for me, whereas later versions aren't
- updated `Gemfile.lock` and `yarn.lock` for compatibility

## To do?

- I added modified icons with different colours - ideally these would be components that just use `currentColor`, but I couldn't work out how to do that in this Rails Vue setup. Any ideas, or shall we leave this for now?
- I also noticed the designs are using Helvetica Neue, but we'd probably need to change that site-wide if we use it?

# Testing
- load server
- navigate to [/criteria/new](localhost:3000/criteria/new)
  -  check navigation against [design](https://xd.adobe.com/view/85f38f01-ca5a-4c6b-a393-28b15a7e278b-d0a0/specs)
  - click 'Exit' button to launch modal and check against [design](https://xd.adobe.com/view/85f38f01-ca5a-4c6b-a393-28b15a7e278b-d0a0/screen/d5e07126-6e55-4030-b926-e60ebd8e3351/specs)
- navigate to [/commitments/new](localhost:3000/commitments/new)
  -  check navigation against [design](https://xd.adobe.com/view/85f38f01-ca5a-4c6b-a393-28b15a7e278b-d0a0/screen/9c7965c3-64f7-46a7-bd7d-8188712d59eb/specs)
  - click 'Exit' button to launch modal and check against [design](https://xd.adobe.com/view/85f38f01-ca5a-4c6b-a393-28b15a7e278b-d0a0/screen/0cac2fea-f68a-41ec-83bf-4c117ff9052e/specs)
- in `/app/javascript/components/form/SurveyForm.vue`, change `isFirstPage` to `false` (keeping `isLastPage` set to `false`) to check against middle page commitment form designs (e.g. [location page](https://xd.adobe.com/view/85f38f01-ca5a-4c6b-a393-28b15a7e278b-d0a0/screen/9de804d6-0cb2-4d31-b51d-80ef1d67d15e/specs))
- in `/app/javascript/components/form/SurveyForm.vue`, change `isFirstPage` to `false` and `isLastPage` to `true` to check against [last page commitment form design](https://xd.adobe.com/view/85f38f01-ca5a-4c6b-a393-28b15a7e278b-d0a0/screen/69cf8c9f-4a32-4f5e-939c-e4474644615f/specs)
